app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,46 @@
+class FriendRequestsController < ApplicationController
+	before_action :set_friend_request, only: [:destroy, :accept]
+
+  def index
+    @friend_requests = current_user.friend_requests_received
+  end
+
+  def create
+    @friend_request = current_user.friend_requests_sent.build(receiver_id: params[:receiver_id])
+    if @friend_request.save
+      respond_to do |format|
+          format.html { redirect_back(fallback_location: root_path, notice: 'Friend request sent.') }
+          format.js
+        end
+    else
+      flash[:error] = 'Friend request could not be sent'
+    end
+  end
+
+  def destroy
+    @friend_request.destroy
+    respond_to do |format|
+      format.html { redirect_to friend_requests_path }
+      format.js
+    end
+  end
+
+  def accept
```

Comment:
There are a couple of things I'd like to point out here:

1. Please see my comment on `users_controller.rb` about placing non-CRUD actions as public methods in a resources controller. 

2. I think we can simplify the logic of a person accepting another user's friend request.

- We can do away with the [`accept`](https://github.com/zinagithub/odin_facebook/blob/25f5b79cc7921d8fe91c251cdd52d2d41c50c497/app/models/friend_request.rb#L10) method in the `friend_request.rb` file
- We should have a `friendships_controller` with a `create` action
- We can have a `after_save` callback in the `friendship.rb` file that will destroy the associated friend_request after the `friendship` record is created.

```ruby
# friendship.rb

after_save :delete_request_associated

private

def delete_request_associated
 # logic that finds the request associated for the friendship record (user_id is sender_id and friend_id is receiver_id) and destroys it
end
```
This way our methods are much simpler, in our `friendships_controller.rb` we can simply have a very very simple create action:

```ruby
def create
 @friendship = current_user.friendships.build(friend_id: params[:friend_id])

 if @friendship.save
   # do something
 else
   # do something else
 end
end
```

While here in `friend_requests_controller.rb` we don't care what is happening in `friendship` anymore. This controller just knows how to `create` and how to `destroy` a `friend_request` record.